### DISCUSSION TITLE
LilyGO T HMI  ESP32  S3   ST7789  Parellel  8   and  XPT2046  SPI  touch  display  confi

### DISCUSSION THREAD

**Discussion by vednildip** [lovyan03/LovyanGFX#646] [Device Configurations]
Since  RD  and  RESET  pin  is not  defined  is -1 in  LilyGO T HMI  ESP32  S3   ST7789  Parellel  8   and  XPT2046  SPI  touch  display   How  to  configure  available  pins 
// lcd
#define LCD_DATA0_PIN (48)
#define LCD_DATA1_PIN (47)
#define LCD_DATA2_PIN (39)
#define LCD_DATA3_PIN (40)
#define LCD_DATA4_PIN (41)
#define LCD_DATA5_PIN (42)
#define LCD_DATA6_PIN (45)
#define LCD_DATA7_PIN (46)
#define PCLK_PIN      (8)
#define CS_PIN        (6)
#define DC_PIN        (7)
#define RST_PIN       (-1)
#define BK_LIGHT_PIN  (38)

// touch screen
#define TOUCHSCREEN_SCLK_PIN (1)
#define TOUCHSCREEN_MISO_PIN (4)
#define TOUCHSCREEN_MOSI_PIN (3)
#define TOUCHSCREEN_CS_PIN   (2)
#define TOUCHSCREEN_IRQ_PIN  (9)

Touch having  compilation  error  : error: 'HSPI_HOST' was not declared in this scope; did you mean 'SPI3_HOST'?
   73 |       cfg.spi_host = HSPI_HOST;// 使用するSPIを選択 (HSPI_HOST or VSPI_HOST   and  when used  SPI3_HOST  code  compiled

**Comment by ottoklein** [comment]
[LGFX_LilyGO_T-HMI.zip](https://github.com/user-attachments/files/19104363/LGFX_LilyGO_T-HMI.zip)
For Lovyan GFX configuration
73 Kurt

**Comment by ottoklein** [comment]
[LilyGo-T-HMI_Arduino_GFX_red_point_Touch_XPT2046.zip](https://github.com/user-attachments/files/19104592/LilyGo-T-HMI_Arduino_GFX_red_point_Touch_XPT2046.zip)
For Arduino GFX INO file with configuration.
73 k.

**Reply by Westcott1** [reply]
Thanks for posting!
I found your config zip 5 minutes **after** I'd worked out the pins by myself.
This is the old 2.8inch 240x320 panel.

**Reply by ottoklein** [reply]
It's always good to have a second opinion (grin).

**Reply by ottoklein** [reply]
//I have attached a calibration for the touchscreen on NVS storage, if you're interested.

#define HRES 320  // Default screen resulution for X axis
#define VRES 240  // Default screen resulution for Y axis


#include <Preferences.h>
Preferences prefs;


uint16_t Calibrate[8];
uint16_t ReStore[8];

void calibration() {
  
  Serial.begin(115200);
  prefs.begin("prefs", false); // Read only = true  // Open our namespace (or create it, if it doesn't exist) in RO mode.
  if (!prefs.isKey("Container")) {  // exist Storage?
    
    tft.fillScreen(TFT_BLUE);
    tft.calibrateTouch(Calibrate, TFT_WHITE, TFT_BLUE, 20);  // Bodmer
    prefs.putBytes("Container", (byte*)(&Calibrate), sizeof(Calibrate));  //put Calibration data into Storage
  }
 prefs.getBytes("Container", &ReStore, sizeof(ReStore));  // get Calibration data from Storage

  //################### Change Loveyan / Bodmer #####################
  tft.setTouchCalibrate(ReStore);  //Lovyan Save Calibration
  //tft.setTouch(ReStore);//Bodmer Save Calibration
  //################################################################

  Serial.print("Calibration Data: {");
  for (byte i = 0; i < 7; i++) { Serial.printf("%d,", ReStore[i]); }
  Serial.printf("%d}\n", ReStore[7]);

/*
  Serial.println(prefs.freeEntries()); //491 Free space
  Serial.println(prefs.getBytesLength("Container"));//16
  Serial.println(prefs.getType("Container"));//9 = Bytes
 */
}

void newCalibration() {
  tft.setTextColor(TFT_WHITE, TFT_BLUE);
  tft.fillScreen(TFT_BLUE);
  tft.setTextSize(1);
  tft.drawCentreString("New calibration NVS!", HRES / 2, 100, 4);

  delay(500);  
  prefs.remove("Container");  // delete only File
  ESP.restart();
}

**Comment by vednildip** [comment]
Thank you for your mail.

Regard
Nilesh

**Reply by ottoklein** [reply]
you're welcome, I hope you enjoy it
Regard
Kurt

**Comment by vednildip** [comment]
the  uploaded  from  library  manager  #include
<XPT2046_Touchscreen_TT.h>//XPT2046_Touchscreen_TT


require adafruit GFX

On Thu, Mar 6, 2025 at 2:25 PM ottoklein ***@***.***> wrote:

> LilyGo-T-HMI_Arduino_GFX_red_point_Touch_XPT2046.zip
> <https://github.com/user-attachments/files/19104592/LilyGo-T-HMI_Arduino_GFX_red_point_Touch_XPT2046.zip>
> For Arduino GFX INO file with configuration.
> 73 k.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lovyan03/LovyanGFX/discussions/646#discussioncomment-12411423>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BCHAUZIWFHCCLGGU2EUUFLL2TAERTAVCNFSM6AAAAABR2RSD2OVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTENBRGE2DEMY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Regard
Nilesh

**Comment by ottoklein** [comment]
//do not forget to turn on power and light in the setup

//https://github.com/lovyan03/LovyanGFX/discussions/646
// 10.02.2025 Board: ESP 3.1.1 - lLibrary:LovyanGFX 1.1.20 ok

#include "LGFX_LilyGO_T-HMI.h"
#include <LGFX_TFT_eSPI.h>

static TFT_eSPI tft;


void setup() {
  pinMode(PWR_EN_PIN, OUTPUT);
  digitalWrite(PWR_EN_PIN, HIGH);
  pinMode(PWR_ON_PIN, OUTPUT);
  digitalWrite(PWR_ON_PIN, HIGH);

  Serial.begin(115200);
  tft.init();
  tft.setBrightness(255);
//.......................................................................
